internal/btf/core.go

```diff
@@ -590,51 +676,68 @@ func coreFindField(local Type, localAcc coreAccessor, target Type) (_, _ coreFie
 			targetMaybeFlex = last
 			targetOffset += targetMember.OffsetBits
 
+			if targetMember.BitfieldSize > 0 {
+				// For bitfields we compute the offset from which to load
+				// the value, and we include the offset of the bitfield and its
+				// size in 'coreField' so we can later compute the proper bit shift.
+				size, offset, err := calculateBitfieldLoadJussi(-1, int(targetOffset), int(targetMember.BitfieldSize))
+				if err != nil {
+					return coreField{}, coreField{}, true, err
+				}
+				target := target.copy()
+				target.(*Int).Size = uint32(size)
```

Comment:
I am not sure to understand the goal of this two lines, can you please explain me them?

Reply:
The target is coming from the spec, which we don't want to mutate, hence the copy. We set the size here to change the load to the potentially smaller load we just computed.

By the way, I'm not yet sure what's the right approach for figuring out the load size and offset. I don't think there's a performance (or any other) argument to be made of for doing smaller loads. The simplest solution might be to just use the load as specified in the target spec: use the same size and compute offset by figuring out what's the offset of the first bitfield member of the variable.